src/PIL/ImageOps.py

```diff
@@ -74,9 +74,15 @@ def autocontrast(image, cutoff=0, ignore=None):
                    high ends. Either a tuple of (low, high), or a single
                    number for both.
     :param ignore: The background pixel value (use None for no background).
+    :param mask: histogram used in contrast operation is computed using pixels
+                 within the mask. If no mask is given the entire image is used
+                 for histogram computation.
     :return: An image.
     """
-    histogram = image.histogram()
+    if mask:
+        histogram = image.histogram(mask)
+    else:
+        histogram = image.histogram()
```

Comment:
Just pass the `mask` in? It defaults to `None`:

`    def histogram(self, mask=None, extrema=None):`

```suggestion
    histogram = image.histogram(mask)
```

Reply:
Agree. good catch.